RecoHI/HiCentralityAlgos/plugins/CentralityProducer.cc

```diff
@@ -245,6 +250,22 @@ namespace reco {
       }
     }
 
+    if (producePF_) {
+      creco->etPFhfSumPlus_ = 0;
+      creco->etPFhfSumMinus_ = 0;
+      for (const auto& pf : iEvent.get(srcPF_)) {
```

Comment:
It would be a bit more efficient just to continue if the candidate does not have `pdgId` 1 or 2. 
I think you cannot have `pdgId` with -1 or -2 such that the abs is not necessary. 
If think you can just do:
`if (pf.eta() > 0){...}   
 else{...}`

Reply:
done in 3a507b0c89d1a866b1c6b9b098f84e2649964377